internal/commands/user/shared/inputs.go

```diff
@@ -0,0 +1,96 @@
+package shared
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/AlecAivazis/survey/v2"
+)
+
+// UsersInputs are the filtering inputs for a user command
+type UsersInputs struct {
+	State             realm.UserState
+	ProviderTypes     []string
+	Status            StatusType
+	InteractiveFilter bool
+}
+
+// ResolveUsers will use the provided Realm client to resolve the users specified by the realm.App through inputs
+func (i *UsersInputs) ResolveUsers(ui terminal.UI, client realm.Client, app realm.App) ([]string, error) {
+	var err error
+
+	userFilter := realm.UserFilter{}
+	if i.Status != StatusTypeNil {
+		userFilter.Pending = i.Status == StatusTypePending
+	}
+	if len(i.ProviderTypes) > 0 {
+		userFilter.Providers = i.ProviderTypes
+	}
+	if i.State != realm.UserStateNil {
+		userFilter.State = i.State
+	}
+	selectableUsersSet := make(map[string]realm.User)
```

Comment:
[mega-nit] we don't gain much w.r.t. the make here and maps. make is best used for slices, maps can just be created inline since we don't create with a len/capacity like we do slices `map[string]realm.User{}`